libcontainer/cgroups/fs/pids.go

```diff
@@ -45,21 +44,18 @@ func (s *PidsGroup) GetStats(path string, stats *cgroups.Stats) error {
 	}
 	current, err := fscommon.GetCgroupParamUint(path, "pids.current")
 	if err != nil {
-		return fmt.Errorf("failed to parse pids.current - %s", err)
+		return err
 	}
 
-	maxString, err := fscommon.GetCgroupParamString(path, "pids.max")
+	max, err := fscommon.GetCgroupParamUint(path, "pids.max")
 	if err != nil {
-		return fmt.Errorf("failed to parse pids.max - %s", err)
+		return err
 	}
-
-	// Default if pids.max == "max" is 0 -- which represents "no limit".
-	var max uint64
-	if maxString != "max" {
-		max, err = fscommon.ParseUint(maxString, 10, 64)
-		if err != nil {
-			return fmt.Errorf("failed to parse pids.max - unable to parse %q as a uint from Cgroup file %q", maxString, filepath.Join(path, "pids.max"))
-		}
+	// If no limit is set, read from pids.max returns "max", which is
+	// converted to MaxUint64 by GetCgroupParamUint. Historically, we
+	// represent "no limit" for pids as 0, thus this conversion.
+	if max == math.MaxUint64 {
+		max = 0
```

Comment:
This hunk isn't right -- the `pids.max` file contains the string `max` if its in unlimited mode and such a string will fail to parse here (the `math.MaxUint64` fallback also isn't necessary I believe).

Reply:
Please re-review. 

The logic is explained in the commit message and in the comment above. We use `fscommon.GetCgroupParamUint` which returns `max.MaxUint64` in case the `max` string is read from the file:
```go
  // GetCgroupParamUint reads a single uint64 value from the specified cgroup file.
  // If the value read is "max", the math.MaxUint64 is returned.
  func GetCgroupParamUint(path, file string) (uint64, error) {
          contents, err := GetCgroupParamString(path, file)
          if err != nil {
                  return 0, err
          }
          contents = strings.TrimSpace(contents)
          if contents == "max" {
                  return math.MaxUint64, nil
          }
 ```
 
 ...and we convert `math.MaxUint64` to `0` as we historically return `0` for "no limit set".

Reply:
For the commit message, see https://github.com/opencontainers/runc/pull/3011/commits/adcd3b4451303fe2061177189be74f491671cd41
(it's sad that github PR review UI downplays individual commits in favor of overall PR changes)

Reply:
Yeah sorry, for PRs like this I do a review on the entire changeset because doing commit-by-commit reviews isn't as convenient as it is when you're doing them in a mail client. Downside is I miss explanations in commit messages. Sorry for the noise (and I didn't know that the `max` handling already existed).

Reply:
I also missed that the tests pass, which they shouldn't if this was a real bug.